cmake/CeleritasUtils.cmake

```diff
@@ -42,16 +42,50 @@ macro(celeritas_find_package_config _package)
   find_package_handle_standard_args(${_package} CONFIG_MODE)
 endmacro()
 
+macro(celeritas_add_library)
+  if(CELERITAS_USE_CUDA)
+     cuda_add_library(${ARGV})
+  else()
+     add_library(${ARGV})
+  endif()
+endmacro()
+
 function(celeritas_link_vecgeom_cuda target)
-  set_target_properties(${target} PROPERTIES
-    LINKER_LANGUAGE CUDA
-    CUDA_SEPARABLE_COMPILATION ON
-  )
+
+#  set_target_properties(${target} PROPERTIES
+#    LINKER_LANGUAGE CUDA
+#    CUDA_SEPARABLE_COMPILATION ON
+#  )
+
+  # Note: the repeat (target name, location) below is due the author of  cuda_add_library_depend
+  # not knowing how to automatically go from the target to the real file from a generator expression in add_custom_command
+  get_property(vecgeom_static_target_location TARGET VecGeom::vecgeomcuda_static PROPERTY LOCATION)
+  cuda_add_library_depend(${target} VecGeom::vecgeom VecGeom::vecgeomcuda_static ${vecgeom_static_target_location})
   target_link_libraries(${target}
     PRIVATE
+    VecGeom::vecgeom
     VecGeom::vecgeomcuda
     VecGeom::vecgeomcuda_static
   )
+  target_link_libraries(${target}_static
+    VecGeom::vecgeom
+    VecGeom::vecgeomcuda_static
+    ${PRIVATE_DEPS}
+  )
+  SET_TARGET_PROPERTIES(${target}_static PROPERTIES LINKER_LANGUAGE CXX)
+
+  GET_TARGET_PROPERTY(target_include_directories ${target} INCLUDE_DIRECTORIES )
+  SET_TARGET_PROPERTIES(${target}_static PROPERTIES INCLUDE_DIRECTORIES "${target_include_directories}" )
+
+  GET_TARGET_PROPERTY(target_interface_include_directories ${target} INTERFACE_INCLUDE_DIRECTORIES )
+  SET_TARGET_PROPERTIES(${target}_static PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${target_interface_include_directories}")
+
+  GET_TARGET_PROPERTY(target_LINK_LIBRARIES ${target} LINK_LIBRARIES )
+  GET_TARGET_PROPERTY(target_INTERFACE_LINK_LIBRARIES ${target} INTERFACE_LINK_LIBRARIES )
+  SET_TARGET_PROPERTIES(${target}_static PROPERTIES LINK_LIBRARIES "${target_LINK_LIBRARIES}" INTERFACE_LINK_LIBRARIES "${target_INTERFACE_LINK_LIBRARIES}")
```

Comment:
Even though this code is (hopefully) temporary, could you update the formatting to match the rest of the cmake?
- Lowercase function calls, 
- 80-character lines,
- separate key/value pairs for `set_target_properties` for readability
- Remove dead code rather than commenting out